### PR TITLE
:dizzy: replace http links with https

### PR DESCRIPTION
**Pull request by mairaw**
Used Duncan's regex to search and replace 
search: `(]\(|href=")http://([a-zA-Z\.]*(microsoft.com|office.com|msdn.com|aka.ms))`
replace: `$1https://$2`

Didn't fix links for offline books, didn't check if they still work. That's for a separate PR.

I did remove though cases where the fwlink was repeated separately such as 
image

Contributes to #4504

### CODE REVIEW

File: docs/csharp/misc/cs3001.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ ms.assetid: c4f3e247-5e47-4182-b415-c573fb1a152f
 # Compiler Warning (level 1) CS3001
 Argument type 'type' is not CLS-compliant  
   
- A [public](../../csharp/language-reference/keywords/public.md), [protected](../../csharp/language-reference/keywords/protected.md), or `protected internal` method must accept a parameter whose type is compliant with the Common Language Specification (CLS). For more information on CLS Compliance, see [Writing CLS-Compliant Code](http://msdn.microsoft.com/library/4c705105-69a2-4e5e-b24e-0633bc32c7f3) and [Language Independence and Language-Independent Components](../../../docs/standard/language-independence-and-language-independent-components.md).  
+ A [public](../../csharp/language-reference/keywords/public.md), [protected](../../csharp/language-reference/keywords/protected.md), or `protected internal` method must accept a parameter whose type is compliant with the Common Language Specification (CLS). For more information on CLS Compliance, see [Writing CLS-Compliant Code](https://msdn.microsoft.com/library/4c705105-69a2-4e5e-b24e-0633bc32c7f3) and [Language Independence and Language-Independent Components](../../../docs/standard/language-independence-and-language-independent-components.md).
```
Comment (rpetrusha) [inline docs/csharp/misc/cs3001.md]: nit: Writing CLS-Compliant code can be deleted. All of its content is contained in the Language Independence topic.

File: docs/csharp/misc/cs3002.md [hunk 2]
```diff
@@ -10,7 +10,7 @@ ms.assetid: 34f19735-76d2-4d78-bd52-45989a86fca4
 # Compiler Warning (level 1) CS3002
 Return type of 'method' is not CLS-compliant  
   
- A [public](../../csharp/language-reference/keywords/public.md), [protected](../../csharp/language-reference/keywords/protected.md), or `protected internal` method must return a value whose type is compliant with the Common Language Specification (CLS). For more information on CLS Compliance, see [Writing CLS-Compliant Code](http://msdn.microsoft.com/library/4c705105-69a2-4e5e-b24e-0633bc32c7f3) and [Language Independence and Language-Independent Components](../../../docs/standard/language-independence-and-language-independent-components.md).  
+ A [public](../../csharp/language-reference/keywords/public.md), [protected](../../csharp/language-reference/keywords/protected.md), or `protected internal` method must return a value whose type is compliant with the Common Language Specification (CLS). For more information on CLS Compliance, see [Writing CLS-Compliant Code](https://msdn.microsoft.com/library/4c705105-69a2-4e5e-b24e-0633bc32c7f3) and [Language Independence and Language-Independent Components](../../../docs/standard/language-independence-and-language-independent-components.md).
```
Comment (rpetrusha) [inline docs/csharp/misc/cs3002.md]: same comment here: Writing CLS-Compliant Code can be removed.

File: docs/csharp/misc/cs3004.md [hunk 3]
```diff
@@ -10,4 +10,4 @@ ms.assetid: 84aa3b44-42b7-4d31-82b8-386e56f88129
...
   
- Only composed UNICODE characters are allowed in [public](../../csharp/language-reference/keywords/public.md), [protected](../../csharp/language-reference/keywords/protected.md), or `protected internal` identifiers in order to be compliant with the Common Language Specification (CLS). For more information on CLS Compliance, see [Writing CLS-Compliant Code](http://msdn.microsoft.com/library/4c705105-69a2-4e5e-b24e-0633bc32c7f3) and [Language Independence and Language-Independent Components](../../../docs/standard/language-independence-and-language-independent-components.md).
+ Only composed UNICODE characters are allowed in [public](../../csharp/language-reference/keywords/public.md), [protected](../../csharp/language-reference/keywords/protected.md), or `protected internal` identifiers in order to be compliant with the Common Language Specification (CLS). For more information on CLS Compliance, see [Writing CLS-Compliant Code](https://msdn.microsoft.com/library/4c705105-69a2-4e5e-b24e-0633bc32c7f3) and [Language Independence and Language-Independent Components](../../../docs/standard/language-independence-and-language-independent-components.md).
```
Comment (rpetrusha) [inline docs/csharp/misc/cs3004.md]: The link to Writing CLS-Compliant Code can be removed.

File: docs/csharp/misc/cs3005.md [hunk 4]
```diff
@@ -10,7 +10,7 @@ ms.assetid: 64b687e3-2dbd-45dd-b6da-81f77eb7d6bd
...
+ A [public](../../csharp/language-reference/keywords/public.md), [protected](../../csharp/language-reference/keywords/protected.md), or `protected internal` identifier, which differs from another `public`, `protected`, or `protected internal` identifier only in the case of one or more letters, is not compliant with the Common Language Specification (CLS). For more information on CLS Compliance, see [Writing CLS-Compliant Code](https://msdn.microsoft.com/library/4c705105-69a2-4e5e-b24e-0633bc32c7f3) and [Language Independence and Language-Independent Components](../../../docs/standard/language-independence-and-language-independent-components.md).
```
Comment (rpetrusha) [inline docs/csharp/misc/cs3005.md]: The link to Writing CLS-Compliant Code can be removed.

File: docs/csharp/misc/cs3006.md [hunk 5]
```diff
@@ -10,7 +10,7 @@ ms.assetid: efbfd898-e46f-4c3d-91e2-e2da0056b016
...
   
- A method does not cannot be overloaded based on the [ref](../../csharp/language-reference/keywords/ref.md) or [out](../../csharp/language-reference/keywords/out-parameter-modifier.md) parameter and still comply with the Common Language Specification (CLS). For more information on CLS Compliance, see [Writing CLS-Compliant Code](http://msdn.microsoft.com/library/4c705105-69a2-4e5e-b24e-0633bc32c7f3) and [Language Independence and Language-Independent Components](../../../docs/standard/language-independence-and-language-independent-components.md).  
+ A method does not cannot be overloaded based on the [ref](../../csharp/language-reference/keywords/ref.md) or [out](../../csharp/language-reference/keywords/out-parameter-modifier.md) parameter and still comply with the Common Language Specification (CLS). For more information on CLS Compliance, see [Writing CLS-Compliant Code](https://msdn.microsoft.com/library/4c705105-69a2-4e5e-b24e-0633bc32c7f3) and [Language Independence and Language-Independent Components](../../../docs/standard/language-independence-and-language-independent-components.md).
```
Comment (rpetrusha) [inline docs/csharp/misc/cs3006.md]: The link to Writing CLS-Compliant Code can be removed.

**Review by rpetrusha — APPROVED** [review]
LGTM, @mairaw. I left a number of comments about removing the link to Writing CLS-Compliant Code (and there are plenty more that I didn't note), but it strikes me that this is best done in a separate PR, which I'll open tomorrow after this PR is merged.

**Comment by mairaw** [comment]
I prefer if we address these issues separately. I also saw issues while reviewing this content but won’t mix the changes.

**Comment by JasonWHowell** [comment]
@mairaw I used your trick with the regex as shown here to help replace known good http/https links . One comment - it doesn't catch links enumerated at the bottom of the articles when the [anchor][url] format is used, so you might rescan in case .Net uses that convention. Thanks, Jason

**Comment by mairaw** [comment]
Good to know @JasonWHowell! We don't use that convention a lot. And I keep looking at CATS scans for links being flagged there as well.

**Pull request merged**